src/oph/ehoks/palaute/tyoelama.clj

```diff
@@ -107,26 +103,23 @@
     (initial-palaute-state-and-reason jakso hoks opiskeluoikeus nil))
   ([jakso hoks opiskeluoikeus existing-herate]
     (cond
-      (palaute/already-initiated? (vec existing-herate))
-      [nil nil :jaksolle-loytyy-jo-herate]
-
-      (jakso-in-the-past? jakso)
-      [:ei-laheteta :loppu :menneisyydessa]
+      (palaute/already-initiated? existing-herate)
+      [nil :yksiloiva-tunniste :jo-lahetetty]
 
       (opiskeluoikeus/in-terminal-state? opiskeluoikeus (:loppu jakso))
-      [:ei-laheteta :opiskeluoikeus-oid :opiskeluoikeus-terminaalitilassa]
+      [:ei-laheteta :opiskeluoikeus-oid :opiskelu-paattynyt]
 
       (osa-aikaisuus-missing? jakso)
-      [:ei-laheteta nil :osa-aikaisuus-puuttuu]
+      [:ei-laheteta :osa-aikaisuustieto :ei-ole]
 
       (fully-keskeytynyt? jakso)
-      [:ei-laheteta nil :tyopaikkajakso-keskeytynyt]
+      [:ei-laheteta :keskeytymisajanjaksot :jakso-keskeytynyt]
 
       (not-any? suoritus/ammatillinen? (:suoritukset opiskeluoikeus))
       [:ei-laheteta :opiskeluoikeus-oid :ei-ammatillinen]
 
       (palaute/feedback-collecting-prevented? opiskeluoikeus (:loppu jakso))
-      [:ei-laheteta :opiskeluoikeus-oid :rahoitusperuste]
+      [:ei-laheteta :opiskeluoikeus-oid :ulkoisesti-rahoitettu]
```

Comment:
Idea: Tuolla `initiate-if-needed!` funktiossa nykyisellään odotetaan, että `initial-palaute-state-and-reason` palauttama `field` (listan toinen arvo) löytyy HOKSin päätasosta, tän perusteella palautetapahtuman lisatiedot-kenttään haetaan tuota kenttää vastaava arvo HOKSista. Osaamisen hankkimistapaan liittyvistä avaimista ei tuohon lisatiedot-kenttään tallennu mitään.

Yks ratkaisu olis varmaan palauttaa tuon `field` avaimen lisäksi tieto, onko tuo `field` HOKSin päätason tiedoissa vai jakson. Tällöin lisatiedot-kenttään saataisiin haettua tiedot oikeasta paikkaa.

Reply:
Tosi hyvä pointti!  Hmm, jos haettaisiin molemmista, ehkä se toimisi?